roles/openshift_version/tasks/main.yml

```diff
@@ -9,24 +9,53 @@
 # be used by default. Users must indicate what they want.
 - fail:
     msg: "Must specify openshift_release or openshift_image_tag in inventory to install origin. (suggestion: add openshift_release=\"1.2\" to inventory)"
-  when: is_containerized | bool and openshift.common.deployment_type == 'origin' and openshift_release is not defined and openshift_image_tag is not defined
+  when:
+  - is_containerized | bool
+  - openshift.common.deployment_type == 'origin'
+  - openshift_release is not defined
+  - openshift_image_tag is not defined
 
 # Normalize some values that we need in a certain format that might be confusing:
 - set_fact:
     openshift_release: "{{ openshift_release[1:] }}"
-  when: openshift_release is defined and openshift_release[0] == 'v'
+  when:
+  - openshift_release is defined
+  - openshift_release[0] == 'v'
 
 - set_fact:
     openshift_release: "{{ openshift_release | string }}"
   when: openshift_release is defined
 
-- set_fact:
-    openshift_image_tag: "{{ 'v' + openshift_image_tag }}"
-  when: openshift_image_tag is defined and openshift_image_tag[0] != 'v' and openshift_image_tag != 'latest'
-
-- set_fact:
-    openshift_pkg_version: "{{ '-' + openshift_pkg_version }}"
-  when: openshift_pkg_version is defined and openshift_pkg_version[0] != '-'
+# Verify that the image tag is in a valid format
+- block:
+
+  # Verifies that when the deployment type is origin the version:
+  # - starts with a v
+  # - Has 3 integers seperated by dots
+  # It also allows for optional trailing data which:
+  # - must start with a dash
+  # - may contain numbers, letters, dashes and dots.
+  - assert:
+      that:
+      - "{{ openshift_image_tag|match('(^v?\\d+\\.\\d+\\.\\d+(-[\\w\\-\\.]*)?$)') }}"
+      msg: "openshift_image_tag must be in the format v#.#.#[-optional.#]. Examples: v1.2.3, v3.5.1-alpha.1"
+    when: openshift.common.deployment_type == 'origin'
+
+  # Verifies that when the deployment type is (openshift|atomic)-enterprise the version:
+  # - starts with a v
+  # - Has at least 2 integers seperated by dots
```

Comment:
@dgoodwin I think this is 2 integers based on the whole v3.4 tagging so that we may pull whatever is the latest 3.4 based on openshift_release but is that used by openshift_image_tag or should openshift_image_tag always be the exact tag?